plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/ReconnectPinChange.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Primetals Technologies Austria GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Michael Oberlehner
+ *     - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.typemanagement.refactoring;
+
+import java.util.EnumSet;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.fordiac.ide.model.commands.change.ReconnectDataConnectionCommand;
+import org.eclipse.fordiac.ide.model.commands.change.ReconnectEventConnectionCommand;
+import org.eclipse.fordiac.ide.model.commands.delete.DeleteConnectionCommand;
+import org.eclipse.fordiac.ide.model.libraryElement.Connection;
+import org.eclipse.fordiac.ide.model.libraryElement.DataConnection;
+import org.eclipse.fordiac.ide.model.libraryElement.ErrorMarkerInterface;
+import org.eclipse.fordiac.ide.model.libraryElement.EventConnection;
+import org.eclipse.fordiac.ide.model.libraryElement.FBNetworkElement;
+import org.eclipse.fordiac.ide.model.libraryElement.IInterfaceElement;
+import org.eclipse.fordiac.ide.typemanagement.refactoring.IFordiacPreviewChange.ChangeState;
+import org.eclipse.gef.commands.Command;
+import org.eclipse.gef.commands.CompoundCommand;
+import org.eclipse.ltk.core.refactoring.RefactoringStatus;
+
+public class ReconnectPinChange extends AbstractCommandChange<FBNetworkElement> implements IFordiacPreviewChange {
+
+	private final EnumSet<ChangeState> state = EnumSet.of(ChangeState.RECONNECT);
+	private final String newName;
+	private final String oldName;
+
+	protected ReconnectPinChange(final URI elementURI, final Class<FBNetworkElement> elementClass, final String newName,
+			final String oldName) {
+		super("Handle connection of : " + oldName, elementURI, elementClass);
+		this.newName = newName;
+		this.oldName = oldName;
+	}
+
+	@Override
+	protected Command createCommand(final FBNetworkElement element) {
+		return new ReconnectPinByName(oldName, newName, element, state);
```

Comment:
Is there a reason to defer the creation of the actual commands to `ReconnectPinByName.execute()` below, instead of just creating a `CompoundCommand` with the actual commands here?

Reply:
You are right, i will change that